### PR TITLE
[WIP] Added Annoying Messages Telling You You're Slow + Movement Speed Calculation Edits

### PR DESCRIPTION
**Pull request by BurgerLUA**
I'm tired of seeing people walking around while they're full on starving, especially when they walk past the kitchen with all this food. People just don't like eating for whatever reason, so I decided to make a thing that punishes users with annoying text almost every time they take a step as well as calculate movement penalties differently by moving the hunger multiplication at the bottom of the code.

# Movement Speed
Currently, if you wore a hardsuit and were starving, the movement speed penalty would be insignificant as the calculation would be (species movement penalty * hunger move penalty) + hardsuit move penalty. This push makes it so the calculation is actually (species movement penalty + hardsuit move penalty) * hunger move penalty. It would make sense that moving in a hardsuit would be twice as hard when you're starving as your muscles would be weak.

# Messages
There's a chance for an annoying message to pop up every time you move without gravity, and this applies to most movement modifications. The base message chance is rare (5% per step) and gets lower or higher depending on how significant the penalty is. I did a lap around the center of the halls with a hardsuit and magboots on and I got at most 1 annoying message per lap. If you're starving, there's a 20% of an annoying message popping up telling you that you're starving each time you take a step.

# Further Justification
Under Roleplaying Characters: Take into account your character's comfort. Wearing hardsuits or internals for a prolonged period of time would be very tiring or uncomfortable. Do not wear these items without good reason. People just don't seem to take into character their character's comfort. Maybe they don't know that they're starving or maybe they just don't care, I think this sort of thing would make them care as it's incredibly distracting, just like being starving in real life is.

# Things that send you notification
(Percentage is based on per step.)
- Low health (5% - 15%, 0 Health Loss to 100 Health Loss)
- Pain (5% - 105%, 0 Pain to 100 Pain)
- Shoes (6% - ~20%, Depends on the shoe's slowdown. Prisoner Legcuffs is 20% per tile.)
- Suit (6% - 10%, Depends on the suit's slowdown. Hardsuits are like 1 to 5.)
- Drowsiness (5%)
- Being Fat (2.5%)
- High Body Temperature (2.5%)
- Hungry (5%)
- Starving (20%)

Everything else that slows you down, such as having a missing limb, I feel does not need a notification.

### CODE REVIEW

**Comment by TheGreatJorge** [comment]
I kind of disagree with the way this is handled.

For example, annoying fat characters just because they are slowed down a bit seems like an overkill to me. And then again, if the point is to make people eat food and not wear hardsuits all the time, it's the kind of thing that punishes the whole group when only part of it did something wrong. Althoght yeah, you can't ensure only the right people will get annoyed, but I'd say this isn't the best way of handling this.

Sure, you would get uncomfortable wearing hardsuit no matter if you are fixing breaches, or going down a hallway. But if the point is to annoy people about it, then I'd guess there is no point doing that at times you are justified to wear one, because in one case it will achieve the person eventually taking the hardsuit off, while in other case, it will simply annoy a person that is stuck in a hardsuit for the time being.

I understand the PR tried to adress it by not annoying you in areas without gravity, but there are quite a lot of other cases this doesn't cover.

So, I'd suggest using system that only starts to notify you after certain ammount of time? Assuming if you have to wear hardsuit, by the time it would start annoying you, you would take it off. Of course there are lots of cases that timer would run out before that, but I guess it's still better then annoying you right away.

And maybe tying it to certain time intervals or tiles walked instead of random chance would help too. Thought, I've got no clue how performant that would be.

I'd agree with the goal, but I don't think I agree with how it's achieved, yeah.

File: code/modules/mob/living/carbon/human/human_movement.dm [hunk 1]
```diff
@@ -53,29 +66,58 @@
...
 
+	// Putting the hunger system here so being starving is MUCH more noticable as it now considers movement speed penalties
+	//Simpler hunger slowdown calculations, this should be a little faster due to no division, and more scaleable
+	if (nutrition < (max_nutrition * 0.4))
+		tally++
+		if(prob(messagechance))
+			messagechance = messagechance*0.5
+			src << "<span class='warning'>Your lack of nutrition makes movement difficult...</span>"
+	else if (nutrition < (max_nutrition * 0.1))
+		tally+= tally*2 //Yell at me if there is a *= function or something
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_movement.dm]: `//Yell at me if there is a *= function or something`

Pretty sure there is `x *= 2`, but that wouldn't be equivalent to what you wrote. `tally = tally + (tally * 2)` != `tally = tally * 2`

**Comment by skull132** [comment]
What Jorge posted, specifically this:
> So, I'd suggest using system that only starts to notify you after certain ammount of time? Assuming if you have to wear hardsuit, by the time it would start annoying you, you would take it off. Of course there are lots of cases that timer would run out before that, but I guess it's still better then annoying you right away.

Makes sense as well. You shouldn't really get pinged for dipping below limit right quick.

**Comment by skull132** [comment]
I changed my mind after playing a round yesterday. The current hunger messages are already annoying enough. I can scarcely believe them getting _anymore_ frequent. And other messages with similar frequency would get in the way too, simply spamming chat is not an elegant solution.

**Comment by BurgerLUA** [comment]
Adding WIP tag until it's less annoying but still serves it's purpose.

**Comment by Lohikar** [comment]
Status on this?

**Comment by BurgerLUA** [comment]
@Lohikar I should probably shelf the project until I get a better sense of coding before I implement cosmetic features like overlays.

**Pull request closed without merging**